src/VirtoCommerce.MarketingModule.Core/Model/DynamicContent/Conditions/CatalogConditions/DynamicContentConditionCategoryIs.cs

```diff
@@ -0,0 +1,18 @@
+using VirtoCommerce.CoreModule.Core.Common;
+using VirtoCommerce.CoreModule.Core.Conditions;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.MarketingModule.Core.Model.DynamicContent.Conditions.CatalogConditions
+{
+    public class DynamicContentConditionCategoryIs : ConditionTree
+    {
+        public string CategoryId { get; set; }
+        public string CategoryName { get; set; }
+
+        public override bool IsSatisfiedBy(IEvaluationContext context)
+        {
+            return context is DynamicContentEvaluationContext dynamicContentContext
+                && (CategoryId?.EqualsInvariant(dynamicContentContext.CategoryId) ?? false);
```

Comment:
It`s look like we dont need nullcheck here. https://github.com/VirtoCommerce/vc-platform/blob/master/src/VirtoCommerce.Platform.Core/Extensions/StringExtensions.cs#L109
```suggestion
                && (CategoryId.EqualsInvariant(dynamicContentContext.CategoryId));
```